src/manage_sql_agent_groups.c

```diff
@@ -436,9 +436,23 @@ delete_agent_group (const char *agent_group_uuid, int ultimate)
            " SELECT %llu, agent_id FROM agent_group_agents WHERE group_id = %llu;",
            trash_id, agent_group);
 
+    /* Update the location of the agent_group in any trashcan tasks. */
+    sql ("UPDATE tasks"
+         " SET agent_group = %llu,"
+         "     agent_group_location = " G_STRINGIFY (LOCATION_TRASH)
+         " WHERE agent_group = %llu"
+         " AND agent_group_location = " G_STRINGIFY (LOCATION_TABLE) ";",
+         trash_id,
+         agent_group);
+
       permissions_set_locations ("agent_group", agent_group, trash_id, LOCATION_TRASH);
       tags_set_locations ("agent_group", agent_group, trash_id, LOCATION_TRASH);
     }
+  else if (agent_group_in_use (agent_group))
```

Comment:
With an "ultimate" delete, this should also have a check if it's in use by hidden tasks (unlike the check I suggested above).
Also, you you should add a `trash_agent_group_in_use` when trying to ultimately delete an agent group that is in the trash (around line 404).